examples/all-clusters-app/all-clusters-common/src/camera-av-stream-delegate-impl.cpp

```diff
@@ -0,0 +1,373 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.h>
+#include <camera-av-stream-delegate-impl.h>
+#include <fstream>
+#include <iostream>
+#include <lib/support/logging/CHIPLogging.h>
+
+#define SNAPSHOT_FILE_PATH "./capture_snapshot_test.jpg"
+#define SNAPSHOT_FILE_RES_WIDTH (168)
+#define SNAPSHOT_FILE_RES_HEIGHT (112)
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::CameraAvStreamManagement;
+using chip::Protocols::InteractionModel::Status;
+
+// Global pointer to overall Camera AVStream Manager implementing the Cluster delegate.
+std::unique_ptr<CameraAVStreamManager> sCameraAVStreamMgrInstance;
+
+// Global pointer to Camera AVStream Mgmt Server SDK cluster;
+std::unique_ptr<CameraAVStreamMgmtServer> sCameraAVStreamMgmtClusterServerInstance;
+
+Protocols::InteractionModel::Status CameraAVStreamManager::VideoStreamAllocate(const VideoStreamStruct & allocateArgs,
+                                                                               uint16_t & outStreamID)
+{
+    outStreamID               = kInvalidStreamID;
+    bool foundAvailableStream = false;
+
+    for (VideoStream & stream : videoStreams)
+    {
+        if (!stream.isAllocated)
+        {
+            foundAvailableStream = true;
+            break;
+        }
+    }
+
+    if (!foundAvailableStream)
+    {
+        return Status::ResourceExhausted;
+    }
+
+    for (VideoStream & stream : videoStreams)
+    {
+        // Fake allocation with just matching codec
+        if (!stream.isAllocated && stream.codec == allocateArgs.videoCodec)
+        {
+            stream.isAllocated = true;
+            outStreamID        = stream.id;
+            return Status::Success;
+        }
+    }
+
+    return Status::Failure;
```

Comment:
Still needs a spec issue and comment pointing to it here that the behavior on codec mismatch is not defined.  FAILURE is not a good error code to be returning in general.

Reply:
https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/11384